blog/src/a-base32-checksum.md

```diff
@@ -0,0 +1,1241 @@
+  const mod97_10Digit = widget.getElementsByClassName('mod97_10Digit')[0];
+  const updateWidget = () => {
+    const input = iso13616Prepare(mod97_10Input.value);
+    mod97_10Validity.textContent = mod97_10Valid(input) ?  'Valid' : 'Invalid';
+    const digits = iso7064Mod97_10(input);
+    mod97_10Digit.textContent = (digits !== digits) ? 'removal of non-alnums' :
+      digits.toString().padStart(2, '0');
+  };
+  mod97_10Input.addEventListener('input', updateWidget);
+  updateWidget();
+
+  const findCollisionBut = widget.getElementsByClassName('findCollision')[0];
+  const collisionOutput = widget.getElementsByClassName('collision')[0];
+  findCollisionBut.addEventListener('click', event => {
+    const other = mod97_10FindCollision(mod97_10Input.value);
+    if (other === undefined) {
+      collisionOutput.value =
+        'No conflicting single substitution found, try adding valid digits.';
+    } else { collisionOutput.value = other; }
+  });
+}
+addEventListener('DOMContentLoaded', initMod97_10Widget);
+</script>
+
+Furthermore, while it technically allows mixed case, the uppercase letter always
+conflicts with its lowercase, essentially making lowercase too risky to
+use. Uppercase all inputs.
+
+On the other hand, MOD 97-10 checksums do not produce letters, which leaves
+some of the information content of the checksum on the table, that could
+otherwise strengthen the checksum.
+
+## A Blank Slate
+
+If we were to redesign the global banking system, how would we improve this?
+
+The goals we want for our imaginary account number are:
+
+1. **Be easy to transmit.** So, text.
+2. **Be short**, to simplify copying.
+3. **Contain enough space to scale globally**. We don't want to run out of
+   account numbers as the population grows. This, along with the previous
+   constraint, means that we need to encode as many bits of information as
+   we can within each character. base64 maybe? That'd be 6 bits per char.
+4. **Minimize human transcription errors**. base64 is notably poor in this
+   respect, because of the 0/O and I/l/1/i visual confusion. On the other
+   hand, base32 has more information density than just digits, but none of
+   the ambiguity. Finally, while base64 contains a dash or a dot that can
+   be cut off by text processors, base32 will be selected as a single word
+   even when double-clicking.
+5. **Allow early detection of errors**. We don't want to inform the user
+   of their mistake asynchronously. So the algorithm should be easy to
+   implement on the user-interface.
+
+Obviously, I am not the first to look into this. Notably, [Bitcoin][base58]
+created a solution for this. How does it work, and how does it fare?
+
+In Bitcoin, the account number is roughly a (cryptographic) hash of the user's
+public key, cut and concatenated with a checksum that also uses a cryptographic
+hash.
+
+The account number is usually represented in text as [base58][]. It is very
```

Comment:
Here I lost context (the fact that you were talking about Bitcoin account numbers). Might be worth making it more obvious that you are talking about bitcoin in this paragraph and the one below about its Checksum. I see two ways of doing this, either intro'ing each paragraph in a more obvious way ("Bitcoin's account numbers" / "Bitcoin's checksum"), or heading all of those paragraph with a title (e.g: "Bictoin's solution... and its shortecomings")

Reply:
I like re-intro'ing each paragraph :)